packages/salesforcedx-vscode-core/src/conflict/metadataCacheService.ts

```diff
@@ -252,21 +258,64 @@ export class MetadataCacheService {
   }
 
   /**
-   * Groups the information in a MetadataCacheResult by component
+   * Groups the information in a MetadataCacheResult by component.
+   * Child components are returned as an array entry unless their parent is present.
    * @param result A MetadataCacheResult
    * @returns An array with one entry per retrieved component, with all corresponding information about the component included
    */
   public static correlateResults(result: MetadataCacheResult): CorrelatedComponent[] {
     const components: CorrelatedComponent[] = [];
 
-    const projectIndex = new Map<string, SourceComponent>();
+    const projectIndex = new Map<string, RecomposedComponent>();
     for (const comp of result.project.components) {
-      projectIndex.set(MetadataCacheService.makeKey(comp.type.name, comp.fullName), comp);
+      const key = MetadataCacheService.makeKey(comp.type.name, comp.fullName);
+      if (comp.parent) {
+        const parentKey = MetadataCacheService.makeKey(comp.parent.type.name, comp.parent.fullName);
+        const parentEntry = projectIndex.get(parentKey);
+        if (parentEntry) {
+          parentEntry.children.set(key, comp);
+        } else {
+          projectIndex.set(parentKey, {
+            children: new Map<string, SourceComponent>().set(key, comp)
+          });
+        }
+      } else {
+        const entry = projectIndex.get(key);
```

Comment:
Reference to the earlier comment. Can we simply do a projectIndex.set(key, sourceComponent) and have no child components?

Reply:
See above. But also sometimes we may have an entry which has children stored but does not have a parent yet. So in that case we still need to check whether the entry exists in order to not overwrite the children.

So in conclusion do you think both fields should be optional? Or leave as is? Or something else?